Include configuration for docker containers, SSL automation and repla…

mofesola opened the pull request:
Replace CouchRiver with Logstash

* Add automated installation for SSL with Let's Encrypt
* Replace deprecated CouchRiver Plugin with Logstash
* Containerise application with docker and docker-compose
* Introduce Nginx as the webserver for production deployment

Resolves #20

adeolabadmus commented:
@jkleinsc, While rivers has been replaced with logstash in the config and build pipeline, search from the frontend doesn't work yet. I'm currently working on this to perfect it.

Cc @mofesola

jkleinsc commented:
@adeolabadmus thanks for the update.  Let me know if you have any questions about how search works in HospitalRun

jkleinsc commented:
@mofesola @adeolabadmus I just tried setting this up on my Mac and it fails with the following error:
ERROR: Service 'nginx' failed to build: lstat conf/hospitalrun-dev.conf: no such file or directory

What OSes has this been tested on?

mofesola commented:
@jkleinsc I have removed the erring line. Apparently, the file existed on my local so I couldn't catch it. It works fine now.

Also, I reverted to v0.9.9 in server routes, because 0.9.11 was throwing an error and exiting the container. Can we look into this?

Error:
`hospitalrun      | > hospitalrun-server@0.9.18 start /usr/src/app
hospitalrun      | > node app.js
hospitalrun      | 
hospitalrun      | /usr/src/app/node_modules/hospitalrun-server-routes/routes/10-couchproxy.js:19
hospitalrun      |         let configsToUpdate = [];
hospitalrun      |         ^^^
hospitalrun      | 
hospitalrun      | SyntaxError: Block-scoped declarations (let, const, function, class) not yet supported outside strict mode
hospitalrun      |     at exports.runInThisContext (vm.js:53:16)
hospitalrun      |     at Module._compile (module.js:373:25)
hospitalrun      |     at Object.Module._extensions..js (module.js:416:10)
hospitalrun      |     at Module.load (module.js:343:32)
hospitalrun      |     at Function.Module._load (module.js:300:12)
hospitalrun      |     at Module.require (module.js:353:17)
hospitalrun      |     at require (internal/module.js:12:17)
hospitalrun      |     at Array.map (native)
hospitalrun      |     at Object.<anonymous> (/usr/src/app/node_modules/hospitalrun-server-routes/index.js:2:63)
hospitalrun      |     at Module._compile (module.js:409:26)
hospitalrun      | 
hospitalrun      | npm info hospitalrun-server@0.9.18 Failed to exec start script
hospitalrun      | npm ERR! Linux 4.9.12-moby
hospitalrun      | npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "start"
hospitalrun      | npm ERR! node v4.4.3
hospitalrun      | npm ERR! npm  v2.15.1
hospitalrun      | npm ERR! code ELIFECYCLE
hospitalrun      | npm ERR! hospitalrun-server@0.9.18 start: `node app.js`
hospitalrun      | npm ERR! Exit status 1
`

jkleinsc commented:
@mofesola can you change to use Node 6.x instead of 4.x?  That will resolve the error you are getting with hospitalrun-server-routes v0.9.11.

mofesola commented:
Great, I'll do that and push

mofesola commented:
This has been updated with the working logstash configuration, and node has been updated to v6
@jkleinsc

jkleinsc commented:
Looks good to me @mofesola.  Thanks for the PR!  I am going to merge it in.
